### PR TITLE
Add compare-trace command

### PR DESCRIPTION
**Pull request by markrtuttle**
This pull request consists of three commits making three changes:

* Modifies the trace summary produced by make-trace on cbmc json output to more closely match the trace summary produced on cbmc json output 
* Adds a compare-trace command to compare two trace summaries (ignoring known discrepancies between cbmc json and xml output).
* Modifies the compare-result command to ignore a minor, known discrepancy between cbmc text and cbmc json/xml output.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cbmc_viewer/tracet.py [hunk 1]
```diff
@@ -510,7 +517,14 @@ def parse_json_step(step, root=None):
               parse_json_function_return if kind == 'function-return' else
               parse_json_location_only if kind == 'location-only' else None)
     if parser is None:
-        raise UserWarning("Unknown json step type: {}".format(kind))
+        # skip uninteresting kinds of steps
+        if kind in ['loop-head']:
```
Comment (angelonakos) [inline cbmc_viewer/tracet.py:521]: Are you expecting this list to grow with more kinds of steps?

File: cbmc_viewer/tracet.py [hunk 2]
```diff
@@ -541,7 +555,7 @@ def parse_json_assignment(step, root=None):
         raise UserWarning("Unknown json assignment type: {}".format(akind))
 
     # &v is represented as {name: pointer, data: v}
-    # NULL is represented as {name: pointer, data:{(basetype *)NULL)}
+    # NULL is represented as {name: pointer, data: {((basetype *)NULL)}
```
Comment (angelonakos) [inline cbmc_viewer/tracet.py:558]: Missing a `}` in the end of the line?

File: tests/compare-trace [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        trace1 = clean_traces(json.load(handle))
+    with open(args.TRACE2, encoding='utf-8') as handle:
+        trace2 = clean_traces(json.load(handle))
+
+    with open('/tmp/trace1.json', "w", encoding='utf-8') as handle:
+        json.dump(trace1, handle, indent=2)
+    with open('/tmp/trace2.json', "w", encoding='utf-8') as handle:
+        json.dump(trace2, handle, indent=2)
+
+    if trace1 != trace2:
```
Comment (angelonakos) [inline tests/compare-trace:68]: Is insertion-order for the keys inside of the `trace1` and `trace2` dictionaries guaranteed to be the same? If not, the comparison might fail. 

Should sorting operations for `trace1["viewer-trace"]["traces"]` and `trace2["viewer-trace"]["traces"]` be added after cleaning?

File: cbmc_viewer/tracet.py (hunk 1 above)
Comment (markrtuttle) [reply cbmc_viewer/tracet.py:521]: I do expect the list to grow over time.

File: tests/compare-trace (hunk 3 above)
Comment (markrtuttle) [reply tests/compare-trace:68]: I'm comparing them as maps, not ordered structures, so order doesn't matter.

File: cbmc_viewer/tracet.py (hunk 2 above)
Comment (markrtuttle) [reply cbmc_viewer/tracet.py]: Done, Thanks.

**Review by angelonakos — APPROVED** [review]

**Pull request merged**